lms/djangoapps/instructor_task/tasks_helper.py

```diff
@@ -1023,6 +1029,47 @@ def upload_may_enroll_csv(_xmodule_instance_args, _entry_id, course_id, task_inp
     return task_progress.update_task_state(extra_meta=current_step)
 
 
+def generate_students_certificates(_xmodule_instance_args, _entry_id, course_id, task_input, action_name):
+    """
+    For a given `course_id`, generate certificates for all students
+    that are enrolled.
+    """
+    start_time = time()
+    enrolled_students = CourseEnrollment.objects.users_enrolled_in(course_id)
+    task_progress = TaskProgress(action_name, enrolled_students.count(), start_time)
+
+    current_step = {'step': 'Calculating students already have certificates'}
+    task_progress.update_task_state(extra_meta=current_step)
+
+    # compute those students where certificates not already generated
+    students_require_certs = [
+        student for student in enrolled_students
+        if certificate_status_for_student(student, course_id)['status'] == CertificateStatuses.unavailable
+    ]
+
+    task_progress.skipped = task_progress.total - len(students_require_certs)
+
+    current_step = {'step': 'Generating Certificates'}
+    task_progress.update_task_state(extra_meta=current_step)
+
+    course = modulestore().get_course(course_id, depth=0)
+    # Generate certificate for each student
+    for student in task_progress.attempted:
```

Comment:
@ormsbee -- this seems like a possible opportunity to parallelize some work -- do you agree?